evm/logic/block.py

```diff
@@ -4,26 +4,26 @@
 def blockhash(computation):
     block_number = computation.stack.pop(type_hint=constants.UINT256)
 
-    block_hash = computation.vm.get_ancestor_hash(block_number)
+    block_hash = computation.state.get_ancestor_hash(block_number)
 
     computation.stack.push(block_hash)
 
 
 def coinbase(computation):
-    computation.stack.push(computation.vm.block.header.coinbase)
+    computation.stack.push(computation.state.coinbase)
 
 
 def timestamp(computation):
-    computation.stack.push(computation.vm.block.header.timestamp)
+    computation.stack.push(computation.state.timestamp)
 
 
 def number(computation):
-    computation.stack.push(computation.vm.block.header.block_number)
+    computation.stack.push(computation.state.number)
```

Comment:
*naming nitpick*

Can we keep the `block_` prefix here so it's explicit that we're referring to a block number?

(does this get funky with the introduction of collations later?)

Reply:
Yes, I will modify it and keep `block_` prefix.

(In current design, we haven't added collation header to `State` yet, but I think it's possible in the future.)